gpt_json/streaming.py

```diff
@@ -0,0 +1,78 @@
+import json
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Generic, Optional, Type, TypeVar
+
+from pydantic import BaseModel, create_model
+
+from gpt_json.transformations import JsonFixEnum, fix_truncated_json
+
+SchemaType = TypeVar("SchemaType", bound=BaseModel)
+
+class StreamEventEnum(Enum):
+    OBJECT_CREATED = "OBJECT_CREATED"
+    KEY_UPDATED = "KEY_UPDATED"
+    KEY_COMPLETED = "KEY_COMPLETED"
+    AWAITING_FIRST_KEY = "AWAITING_FIRST_KEY"
+
+
+@dataclass
+class StreamingObject(Generic[SchemaType]):
+  updated_key: str | None
+  value_change: str | None
+  event: StreamEventEnum
+  partial_obj: SchemaType
+
+  schema_model: Type[SchemaType] = None
+     
+  def __class_getitem__(cls, item):
+    new_cls = super().__class_getitem__(item)
+    new_cls.schema_model = item
+    return new_cls
+
+
+def prepare_streaming_object(schema_model: SchemaType, curr_partial_raw: dict[str, Any], prev_partial: 'StreamingObject[SchemaType]', proposed_event: StreamEventEnum) -> StreamingObject[SchemaType]:
```

Comment:
For the prev_partial typehint, is there a reason it's in a quoted string here? The return type looks the same and seems to validate.